### PR TITLE
refactor: restructure the whole thing

### PR DESCRIPTION
**Pull request by cfanoulis**
### Description of the PR

give veza a bit of btf (read: favna) treatment

### Semver Classification

- [x] This PR only includes documentation or non-code changes.
- [x] This PR fixes a bug and does not change the (intended) API/interface.
- [ ] This PR adds methods or properties to the API/interface.
- [ ] This PR removes or renames methods or properties in the API/interface.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,10 +1,23 @@
-{
-  "extends": "bamboo",
-  "rules": {
-    "no-useless-constructor": 0,
-    "@typescript-eslint/unified-signatures": 0
-  },
-  "globals": {
-    "BigInt": true
-  }
+{
```
Comment (Favna) [inline .eslintrc.json:1]: Check what actually has to be included here. The globals don't have to be probably, `ban-types` and `unified-signatures` possibly neither. same for `no-namespace`. Also possibly `no-empty-function` and with that the entire override.

File: LICENSE.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# The MIT License (MIT)
+
+Copyright © `2019` `Antonio Romàn`
```
Comment (Favna) [inline LICENSE.md:3]:
```suggestion
Copyright © `2018` `Antonio Romàn`
```

File: package.json [hunk 3]
```diff
@@ -1,83 +1,130 @@
...
+						"description": "Total Alerts",
+						"href": "https://img.shields.io/lgtm/alerts/g/kyranet/veza.svg?logo=lgtm&logoWidth=18",
+						"url": "https://lgtm.com/projects/g/kyranet/veza/alerts/"
+				},
+				{
+						"description": "Dependabot Status",
+						"href": "https://api.dependabot.com/badges/status?host=github&repo=kyranet/veza",
+						"url": "https://dependabot.com"
+				}
+		],
```
Comment (Favna) [inline package.json:29]: remove this

File: package.json [hunk 4]
```diff
@@ -1,83 +1,130 @@
...
-    "typescript": "^3.7.4"
-  },
-  "dependencies": {
-    "binarytf": "^2.0.0"
-  }
+		"name": "veza",
+		"version": "1.1.0",
+		"description": "IPC/TCP Networking Utility to connect several processes with great speeds.",
+		"main": "dist/index.js",
+		"readme": "README.md",
```
Comment (Favna) [inline package.json:6]:
```suggestion
```

File: package.json [hunk 5]
```diff
@@ -1,83 +1,130 @@
...
+				}
+		],
+		"scripts": {
+				"build": "tsc -b src",
+				"build:examples": "tsc -b examples",
+				"commit": "git-cz",
+				"coverage": "yarn jest --coverage",
+				"coverage:report": "yarn nyc report --reporter=cobertura",
+				"cz": "git-cz",
+				"docs": "typedoc --excludeNotExported --exclude ./src/utils/**/*.ts --mode file --out ./dist/ ./src/",
```
Comment (Favna) [inline package.json:37]: wat?

File: package.json [hunk 6]
```diff
@@ -1,83 +1,130 @@
...
+						"href": "https://api.dependabot.com/badges/status?host=github&repo=kyranet/veza",
+						"url": "https://dependabot.com"
+				}
+		],
+		"scripts": {
+				"build": "tsc -b src",
+				"build:examples": "tsc -b examples",
+				"commit": "git-cz",
+				"coverage": "yarn jest --coverage",
+				"coverage:report": "yarn nyc report --reporter=cobertura",
```
Comment (Favna) [inline package.json:35]:
```suggestion
```

File: package.json [hunk 7]
```diff
@@ -1,83 +1,130 @@
...
+				"docs": "typedoc --excludeNotExported --exclude ./src/utils/**/*.ts --mode file --out ./dist/ ./src/",
+				"format": "prettier --write {src,test,examples}/**/*.ts",
+				"format:check": "prettier --check {src,test,examples}/**/*.ts",
+				"lint": "eslint --fix src test examples --ext ts",
+				"prepublishOnly": "yarn build",
+				"sversion": "standard-version",
+				"test": "jest",
+				"test:watch": "jest --watch",
+				"update": "yarn upgrade-interactive --latest",
+				"watch": "yarn build -w"
```
Comment (Favna) [inline package.json:46]:
```suggestion
				"watch": "tsc -b src -w",
				"clean": "tsc -b src --clean"
```

File: package.json [hunk 8]
```diff
@@ -1,83 +1,130 @@
...
+		"keywords": [
+				"ipc",
+				"tcp",
+				"socket",
+				"veza",
+				"node-ipc"
+		],
+		"contributors": [
+				"kyranet <kyradiscord@gmail.com>"
+		],
```
Comment (Favna) [inline package.json:61]:
```suggestion
```

File: package.json [hunk 9]
```diff
@@ -1,83 +1,130 @@
...
+				"url": "https://github.com/kyranet/veza/issues"
+		},
+		"homepage": "https://veza.js.org",
+		"dependencies": {
+				"binarytf": "^2.0.0"
+		},
+		"devDependencies": {
+				"@commitlint/cli": "^9.1.1",
+				"@commitlint/config-conventional": "^9.1.1",
+				"@skyra/eslint-config": "^5.3.0",
```
Comment (Favna) [inline package.json:74]:
```suggestion
				"@skyra/eslint-config": "^5.3.1",
```

Just published this

File: src/lib/Client.ts [hunk 10]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/unified-signatures */
```
Comment (Favna) [inline src/lib/Client.ts:1]: Already disabled globally

File: src/lib/Structures/Base/SocketHandler.ts [hunk 11]
```diff
@@ -83,8 +83,7 @@ export abstract class SocketHandler {
 				const entry = this.queue.get(id!);
 				/* istanbul ignore next: Hard to reproduce, this is a safe-guard. */
```
Comment (Favna) [inline src/lib/Structures/Base/SocketHandler.ts:84]: remove all the istanbul ignore lines as we're using v8 coverage

File: test/handshake.test.ts [hunk 12]
```diff
@@ -1,42 +1,42 @@
...
 import { get, createServer } from 'http';
 import { serialize } from 'binarytf';
 import { URL } from 'url';
 import { readFileSync } from 'fs';
 
 let port = 8000;
 
-test('Basic Server', { timeout: 5000 }, async t => {
-	t.plan(5);
+beforeEach;
```
Comment (Favna) [inline test/handshake.test.ts:10]: daheck is this line

File: test/handshake.test.ts [hunk 13]
```diff
@@ -1,42 +1,42 @@
...
 	// Connected
-	t.equal(nodeServer.sockets.size, 0, 'The server should not have a connected client.');
-	t.equal(nodeServer.name, 'Server', 'The server should be named after the node.');
+	expect(nodeServer.sockets.size).toBe(0);
+	expect(nodeServer.name).toBe('Server');
 
 	try {
 		await nodeServer.listen(++port);
-		t.fail('This call should definitely crash.');
+		done.fail('This call should definitely crash.');
```
Comment (Favna) [inline test/handshake.test.ts:28]: done.fail is not a function in Jest, that's an incorrect transformation. Remove it.

File: tsconfig.base.json [hunk 14]
```diff
@@ -0,0 +1,9 @@
+{
+	"extends": "@skyra/eslint-config",
+	"compilerOptions": {
+		"lib": ["ESNext", "DOM"],
```
Comment (Favna) [inline tsconfig.base.json:4]:
```suggestion
		"lib": ["ESNext"],
```

This lib doesn't need DOM

File: tsconfig.base.json [hunk 15]
```diff
@@ -0,0 +1,9 @@
...
+		"target": "ESNext",
```
Comment (Favna) [inline tsconfig.base.json:5]:
```suggestion
		"target": "ES2019",
```

File: tsconfig.base.json [hunk 16]
```diff
@@ -0,0 +1,9 @@
...
+		"module": "ESNext",
```
Comment (Favna) [inline tsconfig.base.json:6]:
```suggestion
```

This needs to be compiled to CommonJS not ESM

File: tsconfig.eslint.json [hunk 17]
```diff
@@ -0,0 +1,8 @@
+{
+	"extends": "./tsconfig.base.json",
+	"compilerOptions": {
+		"allowJs": true,
+		"checkJs": true
+	},
+	"include": ["src", "test", "examples", "jest.config.js", "rollup.config.ts"]
```
Comment (Favna) [inline tsconfig.eslint.json:7]:
```suggestion
	"include": ["src", "test", "examples", "jest.config.js"]
```

There is no rollup config

**Review by Favna — CHANGES_REQUESTED** [review]
please don't blindly copypaste...

File: test/handshake.test.ts [hunk 18]
```diff
@@ -1,42 +1,43 @@
-import { Server, Client, NodeClientOptions, ServerSocketStatus, ClientSocketStatus } from '../dist/index';
-import * as test from 'tape';
-import { create } from '../dist/lib/Util/Header';
+/* eslint-global fail*/
```
Comment (Favna) [inline test/handshake.test.ts:1]: wat?

File: test/handshake.test.ts [hunk 19]
```diff
@@ -45,286 +46,273 @@ test('Basic Socket', { timeout: 5000 }, async t => {
...
+	expect.assertions(1);
+	const [nodeServer, nodeSocket] = await setup(done, ++port, { maximumRetries: 0 });
 	await nodeServer.close();
 
 	nodeSocket.on('connecting', () => {
-		t.fail('The socket should not try to connect.');
-	});
-	nodeSocket.on('disconnect', () => {
-		t.pass('The client successfully disconnected.');
+		done.fail('The socket should not try to connect.');
```
Comment (Favna) [inline test/handshake.test.ts:231]: done.fail is not a function. I said that in previous review as well. Remove it. It's just `done()` though in most cases you shouldn't even need it.

**Review by Favna — CHANGES_REQUESTED** [review]

**Comment by cfanoulis** [comment]
it do be gone

**Pull request closed without merging**